var/spack/repos/builtin/packages/spectre/package.py

```diff
@@ -0,0 +1,189 @@
+    geoscience, plasma physics, nuclear physics, and engineering. It runs at
+    petascale and is designed for future exascale computers.
+
+    SpECTRE is being developed in support of our collaborative Simulating
+    eXtreme Spacetimes (SXS) research program into the multi-messenger
+    astrophysics of neutron star mergers, core-collapse supernovae, and
+    gamma-ray bursts."""
+
+    homepage = "https://spectre-code.com"
+    url = "https://github.com/sxs-collaboration/spectre/archive/v2021.12.15.tar.gz"
+    git = "https://github.com/sxs-collaboration/spectre.git"
+
+    maintainers = ['nilsleiffischer']
+
+    generator = 'Ninja'
+
+    version('develop', branch='develop')
+    version('2022.01.03', sha256='872a0d152c19864ad543ddcc585ce30baaad4185c2617c13463d780175cbde5f')
+    version('2021.12.15', sha256='4bfe9e27412e263ffdc6fcfcb84011f16d34a9fdd633ad7fc84a34c898f67e5c')
+
+    # Configuration variants
+    variant('build_type',
+            default='Release',
+            values=("Release", "Debug", "RelWithDebInfo"),
+            description='CMake build type')
+    variant('executables',
+            values=any_combination_of(
+                'SolvePoisson1D', 'SolvePoisson2D', 'SolvePoisson3D',
+                'SolveElasticity2D', 'SolveElasticity3D', 'SolveXcts',
+            ),
+            description="Executables to install")
+    variant('python', default=False, description="Build Python bindings")
+    variant('doc', default=False, description="Build documentation")
+    # TODO: support installation of executables with shared libs
+    # variant('shared',
+    #         default=False,
+    #         description="Build shared libraries instead of static")
+    variant('pch',
+            default=True,
+            description="Compile with pre-compiled headers")
+    variant('memory_allocator',
+            values=('system', 'jemalloc'),
+            multi=False,
+            default='system',
+            description="Which memory allocator to use")
+    variant('formaline',
+            default=True,
+            description=("Write the source tree into simulation output files "
+                         "to increase reproducibility of results"))
+    variant('profiling',
+            default=False,
+            description="Enable options to make profiling SpECTRE easier")
+
+    # Compiler support
+    conflicts('%gcc@:6')
+    conflicts('%clang@:7')
+    conflicts('%apple-clang@:10')
+
+    # Build dependencies
+    depends_on('cmake@3.12:3', type='build')
+    depends_on('ninja', type='build')
+    depends_on('python@2.7:3.10', type='build')
+
+    # Link dependencies
+    depends_on('charmpp@6.10.2:7.0')
+    depends_on('blaze@3.8')
+    depends_on('boost@1.60:1.77')
+    depends_on('brigand@master')
+    depends_on('gsl')
+    depends_on('hdf5@:1.12~mpi')
+    depends_on('jemalloc', when='memory_allocator=jemalloc')
+    depends_on('libsharp@1.0.0:1.0~mpi~openmp')
```

Comment:
Does the code break with `libsharp +mpi` or `+openmp`? If not, then don't put these constraints here.